internal/backend/storage/fossilfs/fossil.go

```diff
@@ -293,6 +328,27 @@ func (f *Fossil) Push(ctx context.Context, remote, branch string) error {
 	return f.PushPull(ctx, "push", remote, branch)
 }
 
+// TryPush calls Push and returns nil if the Fossil repo was not initialized.
+func (f *Fossil) TryPush(ctx context.Context, remote, branch string) error {
+	err := f.Push(ctx, remote, branch)
+	if err == nil {
+		return nil
+	}
+
+	switch err {
+	case store.ErrGitNotInit:
+		debug.Log("Git not initialized. Ignoring.")
+
+		return nil
+	case store.ErrGitNoRemote:
+		debug.Log("Git has no remote. Ignoring.")
```

Comment:
I guess these should read more like:
```suggestion
		debug.Log("Fossil not initialized. Ignoring.")

		return nil
	case store.ErrGitNoRemote:
		debug.Log("Fossil has no remote. Ignoring.")
```

And I'm not sure if the `store.ErrGitNoRemote` and `ErrGitNotInit` actually match any of the Fossil errors?

Reply:
The errors are oddly named (i.e. we should rename them) but the semantics match Fossil, e.g. as Fossil repo can be uninitialized or there could be nothing to commit ....